File: python/preliminar/src/app/common/helpers/bcrypt.py
import bcrypt

def encrypt_password(password: str) -> str:
    salts = bcrypt.gensalt(12)
    return bcrypt.hashpw(password.encode('utf-8'), salts).decode('utf-8')


def is_password_correct(password_to_compare: str, encrypted_password: str) -> bool:
    return bcrypt.checkpw(password_to_compare.encode('utf-8'), encrypted_password.encode('utf-8'))

File: go/gin/internal/common/helper/bcrypt.go
package helper

import (
	"golang.org/x/crypto/bcrypt"
)

func EncryptPassword(password string) (string, error) {
	salts, err := bcrypt.GenerateFromPassword([]byte(password), 12)
	if err != nil {
		return "", err
	}
	return string(salts), nil
}

func IsPasswordCorrect(passwordToCompare, encryptedPassword string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(encryptedPassword), []byte(passwordToCompare))
	return err == nil
}
